aten/src/ATen/native/cuda/AdaptiveAveragePooling.cu

```diff
@@ -512,17 +512,15 @@ namespace {
         AT_ASSERT(input_.numel() < std::numeric_limits<int32_t>::max());
         AT_DISPATCH_FLOATING_TYPES_AND2(kHalf, kBFloat16,
             input_.scalar_type(), "adaptive_avg_pool2d_nhwc_cuda", [&] {
-              AT_SKIP_BFLOAT16_IF_NOT_ROCM(scalar_t, "adaptive_avg_pool2d_nhwc_cuda", [&] {
-                size_t shmem_size = (kernel_size_C * block_x * block_y * block_z) * sizeof(scalar_t);
-                AT_ASSERT(shmem_size <= sharedMemPerBlock);
-                adaptive_average_pool_nhwc<int32_t><<<grid, block, shmem_size, at::cuda::getCurrentCUDAStream()>>> (
-                  input_.data_ptr<scalar_t>(),
-                  output.data_ptr<scalar_t>(),
-                  sizeB, sizeC, isizeH, isizeW, osizeH, osizeW,
-                  kernel_stride_C, kernel_size_C,
-                  istrideB, istrideC, istrideH, istrideW);
-                });
-              }
+              size_t shmem_size = (kernel_size_C * block_x * block_y * block_z) * sizeof(scalar_t);
```

Comment:
lol, all this is never used (adaptive average pooling always goes through fast path, when it does not, and goes here, it produces wrong gradients #44662